source/ui/Components/Output/SevenSegment/SevenSegmentDigit.qml

```diff
@@ -0,0 +1,320 @@
+
+/*
+ * C++ Assembler Interpreter
+ * Copyright (C) 2016 Chair of Computer Architecture
+ * at Technical University of Munich
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import QtQuick 2.0
+import QtQuick.Dialogs 1.0
+
+/*
+Represent one digit of a 7-segment display:
+   _
+  |_|
+  |_|
+
+*/
+
+Rectangle {
+    id: digitBackground
+
+    // Thickness of each segment
+    property var thickness: 20
+    // Background color for the rectangular region around the segments
+    property alias backgroundColor: digitBackground.color
+    // Color of the segments when active.
+    property color digitActiveColor: "yellow"
+    // Color of the segments when inactive.
+    property color digitInactiveColor: "black"
+    // Color of the text inside the segments
+    property color textColor: "gray"
+
+    // Determines whether the corresponding segment of the digit is activated.
+    property bool segment0Activated: false
+    property bool segment1Activated: false
+    property bool segment2Activated: false
+    property bool segment3Activated: false
+    property bool segment4Activated: false
+    property bool segment5Activated: false
+    property bool segment6Activated: false
+
+    // Sent when one of the digit's segments was pressed.
+    signal segmentPressed(var segmentIndex)
+
+    color: "#00000000"
+
+    /*
+      Segments:
+      Will fit their sizes to the parents dimensions.
+      Displays the index of the digit in a byte that represent the segment's state.
+     */
+
+    // Segment 0 (middle-top)
+    SevenSegmentSegment {
```

Comment:
That would be quite difficult as almost each edge has varying properties (x, y, width, height, Text etc.)